api/src/org/apache/cloudstack/api/command/admin/guest/ListGuestOsMappingCmd.java

```diff
@@ -44,10 +44,10 @@
     /////////////////////////////////////////////////////
 
     @Parameter(name = ApiConstants.ID, type = CommandType.UUID, entityType = GuestOsMappingResponse.class, required = false, description = "list mapping by its UUID")
-    private Long id;
+    private String id;
```

Comment:
This can cause issues here, during Cmd object creation arguments values of keys ID and OS_TYPE_ID will be converted to a Long as the parameter type defined here is CommandType.UUID. 

This is where this happens: https://github.com/apache/cloudstack/blob/master/server/src/com/cloud/api/dispatch/ParamProcessWorker.java#L318

If you really want Strings here, please change the @Parameter annotation type to CommandType.STRING etc.